### ISSUE TITLE
[Bug]: After Milvus recovers from a fault where its ETCD/Coord pod is killed, some cases hang and some cases time out during loading.

### ISSUE THREAD

**Issue by zhuwenxing** [milvus-io/milvus#36554]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version:master-20240925-aee046e5-amd64
- Deployment mode(standalone or cluster):cluster
- MQ type(rocksmq, pulsar or kafka):    
- SDK version(e.g. pymilvus v2.0.0rc2):
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

```
[2024-09-25T21:18:41.872Z] [2024-09-25 21:11:12 - DEBUG - ci_test]: (api_request)  : [Collection.load] args: [None, 1, 180], kwargs: {} (api_request.py:62)

[2024-09-25T21:18:41.872Z] [2024-09-25 21:11:12 - DEBUG - ci_test]: (api_response) : None  (api_request.py:37)

[2024-09-25T21:18:41.872Z] [2024-09-25 21:11:12 - DEBUG - ci_test]: (api_request)  : [Collection.search] args: [[[0.10792514621036765, 0.009350127070585424, 0.1431266389420661, 0.05717295188982092, 0.04132065337973011, 0.12256977618633046, 0.0209614588845854, 0.03419539380967773, 0.09551548629235887, 0.013204025470380158, 0.05019446323089156, 0.05020825055519311, 0.15099149023101444, 0.09012002221969224, 0.1......, kwargs: {} (api_request.py:62)

[2024-09-25T21:18:41.872Z] [2024-09-25 21:11:12 - DEBUG - ci_test]: (api_response) : data: ["['id: -7048794156196854212, distance: 22.39752960205078, entity: {}']"]  (api_request.py:37)

[2024-09-25T21:18:41.872Z] [2024-09-25 21:11:12 - INFO - ci_test]: assert search: 0.04822826385498047 (test_all_collections_after_chaos.py:99)

[2024-09-25T21:18:41.872Z] [2024-09-25 21:11:12 - DEBUG - ci_test]: (api_request)  : [Collection.query] args: ['int64 in [-3000, -2999, -2998, -2997, -2996, -2995, -2994, -2993, -2992, -2991, -2990, -2989, -2988, -2987, -2986, -2985, -2984, -2983, -2982, -2981, -2980, -2979, -2978, -2977, -2976, -2975, -2974, -2973, -2972, -2971, -2970, -2969, -2968, -2967, -2966, -2965, -2964, -2963, -2962, -2961, -2960, -......, kwargs: {} (api_request.py:62)

[2024-09-25T21:18:41.872Z] [2024-09-25 21:11:13 - DEBUG - ci_test]: (api_response) : data: ["{'int64': -3000}", "{'int64': -2999}", "{'int64': -2998}", "{'int64': -2997}", "{'int64': -2996}", "{'int64': -2995}", "{'int64': -2994}", "{'int64': -2993}", "{'int64': -2992}", "{'int64': -2991}"] ...  (api_request.py:37)

[2024-09-25T21:18:41.872Z] [2024-09-25 21:11:13 - INFO - ci_test]: assert query result 2000: 0.6076226234436035 (test_all_collections_after_chaos.py:106)

[2024-09-25T21:18:41.872Z] [2024-09-25 21:11:13 - DEBUG - ci_test]: (api_request)  : [Collection.release] args: [], kwargs: {'timeout': 180} (api_request.py:62)

[2024-09-25T21:18:41.872Z] [2024-09-25 21:11:13 - DEBUG - ci_test]: (api_response) : None  (api_request.py:37)

[2024-09-25T21:18:41.872Z] [2024-09-25 21:11:18 - DEBUG - ci_test]: (api_request)  : [Collection.insert] args: [[[-7616306996794355881, -1135490428650009852, -7745712909874256946, -1102073420255682097, -4048776033571265139, 97082861182553775, 7670445939631645318, -8238107405859656038, -5906875047400954443, 4755243609312892806, 6271196569333922483, -167857603213884847, 6879445961164380583, 1396734721482551960......, kwargs: {'timeout': 180} (api_request.py:62)

[2024-09-25T21:18:41.872Z] [2024-09-25 21:11:19 - DEBUG - ci_test]: (api_response) : (insert count: 2000, delete count: 0, upsert count: 0, timestamp: 452800984904894418, success count: 2000, err count: 0  (api_request.py:37)

[2024-09-25T21:18:41.872Z] [2024-09-25 21:11:19 - DEBUG - ci_test]: (api_request)  : [Collection.load] args: [None, 1, 180], kwargs: {} (api_request.py:62)

[2024-09-25T21:18:41.872Z] [2024-09-25 21:14:19 - ERROR - pymilvus.decorators]: RPC error: [wait_for_loading_collection], <MilvusException: (code=1, message=wait for loading collection timeout, collection: Checker__rScxj4za)>, <Time:{'RPC start': '2024-09-25 21:11:19.496412', 'RPC error': '2024-09-25 21:14:19.667079'}> (decorators.py:140)

[2024-09-25T21:18:41.872Z] [2024-09-25 21:14:19 - WARNING - pymilvus.decorators]: Retry timeout: 180s (decorators.py:106)

[2024-09-25T21:18:41.872Z] [2024-09-25 21:14:19 - ERROR - pymilvus.decorators]: RPC error: [load_collection], <MilvusException: (code=1, message=Retry timeout: 180s, message=wait for loading collection timeout, collection: Checker__rScxj4za)>, <Time:{'RPC start': '2024-09-25 21:11:19.273484', 'RPC error': '2024-09-25 21:14:19.667800'}> (decorators.py:140)

[2024-09-25T21:18:41.872Z] [2024-09-25 21:14:19 - ERROR - ci_test]: Traceback (most recent call last):

[2024-09-25T21:18:41.872Z]   File "/home/jenkins/agent/workspace/tests/python_client/utils/api_request.py", line 32, in inner_wrapper

[2024-09-25T21:18:41.872Z]     res = func(*args, **_kwargs)

[2024-09-25T21:18:41.872Z]   File "/home/jenkins/agent/workspace/tests/python_client/utils/api_request.py", line 63, in api_request

[2024-09-25T21:18:41.872Z]     return func(*arg, **kwargs)

[2024-09-25T21:18:41.872Z]   File "/usr/local/lib/python3.8/dist-packages/pymilvus/orm/collection.py", line 429, in load

[2024-09-25T21:18:41.872Z]     conn.load_collection(

[2024-09-25T21:18:41.872Z]   File "/usr/local/lib/python3.8/dist-packages/pymilvus/decorators.py", line 141, in handler

[2024-09-25T21:18:41.872Z]     raise e from e

[2024-09-25T21:18:41.872Z]   File "/usr/local/lib/python3.8/dist-packages/pymilvus/decorators.py", line 137, in handler

[2024-09-25T21:18:41.872Z]     return func(*args, **kwargs)

[2024-09-25T21:18:41.872Z]   File "/usr/local/lib/python3.8/dist-packages/pymilvus/decorators.py", line 176, in handler

[2024-09-25T21:18:41.872Z]     return func(self, *args, **kwargs)

[2024-09-25T21:18:41.872Z]   File "/usr/local/lib/python3.8/dist-packages/pymilvus/decorators.py", line 107, in handler

[2024-09-25T21:18:41.872Z]     raise MilvusException(

[2024-09-25T21:18:41.872Z] pymilvus.exceptions.MilvusException: <MilvusException: (code=1, message=Retry timeout: 180s, message=wait for loading collection timeout, collection: Checker__rScxj4za)>

[2024-09-25T21:18:41.872Z]  (api_request.py:45)

[2024-09-25T21:18:41.872Z] [2024-09-25 21:14:19 - ERROR - ci_test]: (api_response) : <MilvusException: (code=1, message=Retry timeout: 180s, message=wait for loading collection timeout, collection: Checker__rScxj4za)> (api_request.py:46)
```
![image](https://github.com/user-attachments/assets/613b0d14-8cdc-4679-a5b0-99a8324ea7b3)

### Milvus Log

failed job: https://qa-jenkins.milvus.io/blue/organizations/jenkins/chaos-test-straming-node-cron/detail/chaos-test-straming-node-cron/110/pipeline
log:
[artifacts-etcd-pod-kill-110-server-logs.tar.gz](https://github.com/user-attachments/files/17147978/artifacts-etcd-pod-kill-110-server-logs.tar.gz)

**Comment by zhuwenxing** [comment]
/assign @chyezh 
PTAL

**Comment by chyezh** [comment]
ETCD session watcher cannot recover from `ErrCompacted`.

```
[2024/09/25 20:54:20.598 +00:00] [WARN] [resolver/resolver_with_discoverer.go:147] ["service discover break down"] [scheme=milvus-session] [error="etcdserver: mvcc: required revision has been compacted"] [retryInterval=1s]
```

**Comment by chyezh** [comment]
@zhuwenxing should be fixed, please verify it at commit: bd7910632ad5eb58fca10f15a1783f8a1a1fcd10

**Comment by zhuwenxing** [comment]
verified and fixed in `master-20241008-ca9842e4-amd64`
